sp_mock/frontend/src/views/LoginRegister.vue

```diff
@@ -116,6 +121,24 @@ const loginWithLayer8Popup = async () => {
     }
   });
 }
+
+const uploadFile = async (e) => {
+  e.preventDefault();
+
+  const file = e.target.files[0];
+  await layer8.fetch(BackendURL + "/api/upload", {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/layer8.buffer+json",
+    },
+    body: {
+      name: file.name,
+      size: file.size,
+      type: file.type,
+      buff: await file.arrayBuffer(),
+    },
+  });
+};
```

Comment:
An example of loading file with vueJs


```js

const uploadFile = () => {
  if (!selectedFile.value) {
    alert('Please select a file first')
    return
  }

  // Perform the file upload logic here, for example using Axios or Fetch API
  // Replace the placeholder URL with your actual upload endpoint
  const options = ['/upload', '/upload/natif', '/api/upload']
  const uploadEndpoint = 'http://localhost:3000'+ options[2]
  const formData = new FormData()
  formData.append('file', selectedFile.value)

  // Example using Fetch API
  fetch(uploadEndpoint, {
    method: 'POST',
    body: formData
  })
    .then(response => response.json())
    .then(data => {
      // Handle the response from the server
      console.log('File uploaded successfully:', data)
    })
    .catch(error => {
      console.error('Error uploading file:', error)
    })
}
```

Reply:
Yes, but now with us, we wouldn't use `FormData` anymore. We'll handle that internally. So this would now look like:

```js
const uploadFile = () => {
  if (!selectedFile.value) {
    alert('Please select a file first')
    return
  }

  // Perform the file upload logic here, for example using Axios or Fetch API
  // Replace the placeholder URL with your actual upload endpoint
  const options = ['/upload', '/upload/natif', '/api/upload']
  const uploadEndpoint = 'http://localhost:3000'+ options[2]

  // Example using Fetch API
  layer8.fetch(uploadEndpoint, {
    method: 'POST',
    body: {
      name: selectedFile.value.name,
      size: selectedFile.value.size,
      type: selectedFile.value.type,
      buff: await selectedFile.value.arrayBuffer(),
    }
  })
    .then(response => response.json())
    .then(data => {
      // Handle the response from the server
      console.log('File uploaded successfully:', data)
    })
    .catch(error => {
      console.error('Error uploading file:', error)
    })
}
```

Reply:
In this regard, I have to agree with Hermann. Internally, we can fudge the data transfer any way we want, but externally, if a user wants to send multipart form data, they must _feel_ like they are sending multipart form data using a native fetch(...) call.

Reply:
Yes, that's true. That's what I had tried at first, but I was having trouble getting the data bytes. It will be changed later on. I thought this was something we could start with
As regards serving the image, I think that's not a problem at all. I'm almost done getting an example out